packages/ra-core/src/form/FilterLiveForm.tsx

```diff
@@ -111,7 +112,7 @@ export const FilterLiveForm = (props: FilterLiveFormProps) => {
             return;
         }
         formChangesPending.current = true;
-        setFilters(merge({}, filterValues, values));
+        setFilters(mergeObjNotArray(filterValues, values));
```

Comment:
Can't you just use [`Array.isArray`](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Global_Objects/Array/isArray)?